packages/drawer/src/ux-modal-service.ts

```diff
@@ -178,28 +202,47 @@ export class ModalService {
     const bindingContext: ModalBindingContext = {};
     const element = this.createDrawerElement(options, bindingContext);
     
-    element.innerHTML = `<compose view-model.ref=composeViewModel>`;
     if (!options.host || options.host === 'body') {
       options.host = document.body;
     } else if (typeof options.host === 'string') {
       options.host = document.querySelector(options.host) || document.body;
     }
     options.host.appendChild(element);
     let childView = this.templatingEngine.enhance({ element: element, bindingContext: bindingContext });
-    
-    if (options.viewModel && bindingContext.composeViewModel) {
-      bindingContext.composeViewModel.viewModel = options.viewModel;
-    }
-    if (options.view && bindingContext.composeViewModel) {
-      bindingContext.composeViewModel.view = options.view;
-    }
-    if (options.model && bindingContext.composeViewModel) {
-      bindingContext.composeViewModel.model = options.model;
-    }
 
+    let slot: ViewSlot;
     const controllers = (childView as any).controllers as Controller[];
-    
     const drawer: UxDrawer =  controllers[0].viewModel as UxDrawer;
+    try {
+      const view: any = controllers[0].view;
+      slot = new ViewSlot(view.slots['__au-default-slot-key__'].anchor, false);
+    } catch (_error) {
+      this.cancelOpening(drawer);
+      throw 'Missing slot in drawer';
+    }
+
+    let compositionContext = this.createCompositionContext(childView.container as any, element, bindingContext, {
+      viewModel: options.viewModel,
+      view: options.view,
+      model: options.model
+    }, slot);
+
+    compositionContext = await this.ensureViewModel(compositionContext);
+
+    try {
+      const canActivate = await invokeLifecycle(compositionContext.viewModel, 'canActivate', options.model);
+      if (!canActivate) {
+        throw new Error('Drawer cannot be opened');
+      }
+    } catch (error) {
+      this.cancelOpening(drawer);
+      throw error;
+    }
+
+    this.compositionEngine.compose(compositionContext).then((controller) => {
```

Comment:
Should this be awaited?

Reply:
I have decided not to `await` here because the resolve of the compose is not required for the `open` to resolve properly. I thought that this could go on asynchronously. I reckon it wouldn't impact much to `await` if there is a reason why it would be better ?